api/cloud/oc_cloud_manager.c

```diff
@@ -103,13 +147,15 @@ reconnect(void *data)
 static bool
 is_refresh_token_retry_over(oc_cloud_context_t *ctx)
 {
-  return ctx->retry_refresh_token_count >= MAX_RETRY_COUNT;
+  return ctx->retry_refresh_token_count >= MAX_RETRY_COUNT ||
+         g_retry_timeout[ctx->retry_refresh_token_count] == 0;
```

Comment:
I've added a setter (cloud_set_retry) that can change the retry configuration. I use it in one test case. The idea is that you either fill the whole array (max. length MAX_RETRY_COUNT = 6) or you can use less values and 0 will be used as a terminating value (i.e. for example if you want only two values `[2, 4]`, internally the it will be stored in the configuration array as `[2, 4, 0, 0, 0, 0]` so when a 0 is encountered I know that we're at the end; this seems OK for internal use; alternative would be to add a variable to keep track of the size.)